patches/server/1032-Optimise-nearby-player-retrieval.patch

```diff
@@ -23,6 +23,39 @@ index 47e8a6b28960883da868061ed649152ea792c184..f502b01b564bd33c449cbe621966ef40
 +        return this.getNearbyEntities(Player.class, targetPredicate, entity, box);
 +    }
 +
++    public List<ServerPlayer> getNearbyPlayers(double sourceX, double sourceY, double sourceZ, double range, @Nullable Predicate<Entity> targetPredicate) {
++        List<ServerPlayer> nearbyPlayers = new ArrayList<>();
++
++        io.papermc.paper.util.player.NearbyPlayers players = this.chunkSource.chunkMap.getNearbyPlayers();
++
++        com.destroystokyo.paper.util.maplist.ReferenceList<ServerPlayer> nearby = players.getPlayersByBlock(
++            io.papermc.paper.util.CoordinateUtils.getBlockCoordinate(sourceX),
++            io.papermc.paper.util.CoordinateUtils.getBlockCoordinate(sourceZ),
++            io.papermc.paper.util.player.NearbyPlayers.NearbyMapType.GENERAL
++        );
++
++        if (nearby == null) return nearbyPlayers;
++
++        double rangeSquared = range * range;
++
++        Object[] rawData = nearby.getRawData();
++        for (int i = 0, len = nearby.size(); i < len; ++i) {
++            ServerPlayer player = (ServerPlayer)rawData[i];
++
++            if (range >= 0.0) {
```

Comment:
This could just be moved to the top ye ?
if the range is negative, just yield the entire list of players this iterates over/in the world/in the server.

Reply:
Yup, I've changed it to replicate the original getNearestPlayer function.